File: hello-dart/lib/src/logger.dart
import 'package:intl/intl.dart';

/// Simple logging utility
class Logger {
  static final _formatter = DateFormat('yyyy-MM-dd HH:mm:ss.SSS');
  static bool _verboseEnabled = false;

  static void setVerbose(bool enabled) {
    _verboseEnabled = enabled;
  }

  static void debug(String message) {
    if (_verboseEnabled) {
      _log('debug', message);
    }
  }

  static void info(String message) {
    _log('info', message);
  }

  static void warn(String message) {
    _log('warn', message);
  }

  static void error(String message) {
    _log('error', message);
  }

  static void _log(String level, String message) {
    final timestamp = _formatter.format(DateTime.now());
    print('[$timestamp] [$level] $message');
  }
}

File: hello-kotlin/src/main/kotlin/Logger.kt
import java.time.LocalDateTime
import java.time.format.DateTimeFormatter

/**
 * Simple logging utility
 */
object Logger {
    private val formatter = DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm:ss.SSS")
    private var verboseEnabled = false
    
    fun setVerbose(enabled: Boolean) {
        verboseEnabled = enabled
    }
    
    fun debug(message: String) {
        if (verboseEnabled) {
            log("debug", message)
        }
    }
    
    fun info(message: String) {
        log("info", message)
    }
    
    fun warn(message: String) {
        log("warn", message)
    }
    
    fun error(message: String) {
        log("error", message)
    }
    
    private fun log(level: String, message: String) {
        val timestamp = LocalDateTime.now().format(formatter)
        println("[$timestamp] [$level] $message")
    }
}
